app/services/ui_notifications/cache_handler.rb

```diff
@@ -7,7 +7,7 @@ def initialize(user_id = User.current.try(:id))
 
     # JSON Payload
     def payload
-      result = cache.read(cache_key)
+      result = cache.read(cache_key, raw: true)
```

Comment:
Reading https://guides.rubyonrails.org/caching_with_rails.html#low-level-caching it makes me wonder if the following code wouldn't be the same:
```ruby
cache.fetch(cache_key, raw: true, expires_in: cache_expiry) do
  { notifications: notifications }.to_json
end
```

You don't have the debug info about cache hits and misses, but I wonder how many people care about that. I'd just trust the fetch API to do the right thing.

https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-fetch is the exact API for that.

Reply:
And just after I post it I think I realize the reason: expiry does a DB call.

Now that I think about it more: this has been on the agenda to replace with Action Cable: https://edgeguides.rubyonrails.org/action_cable_overview.html can use websockets to keep the connection open. It would mean you no longer need to poll, which reduces the need to cache. We still do a lot of full page loads so it's not completely gone, but helps a lot. Probably an interesting area for optimization.

Reply:
Good find, I'm not sure all that debug info is necessarily worth it either vs. leaning into using the conventions outlined by Rails guide itself.